spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/DatabaseClientTransactionManagerAutoConfiguration.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.spanner;
+
+import com.google.cloud.spanner.DatabaseClient;
+
+import org.springframework.boot.autoconfigure.AutoConfigureOrder;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnSingleCandidate;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.data.spanner.core.SpannerTransactionManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.Ordered;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+/**
+ * Auto-configuration for {@link SpannerTransactionManager}.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+@Configuration
+@EnableTransactionManagement
+@ConditionalOnClass(SpannerTransactionManager.class)
```

Comment:
Like I said before, we really want it to be conditional on something in `spring-tx`. I know there's an issue with that now, but are we tracking it?

Reply:
I don't really understand... Isn't it only conditional if spring-tx is optional and it isn't optional right now? I already stated why we have spring-tx as non-optional (we are using org.springframework.dao.dataaccessexception). We can just not have our exception class extend that one and remove direct use of spring-tx altogether. 

https://github.com/spring-cloud/spring-cloud-gcp/issues/1059 Here is an issue for discussing that. 

Is there something so wrong with including spring-tx all the time? This seems like a core feature we're offering..

Reply:
That issue is what I was looking for. Thanks!